### PR TITLE
[434] Hiring staff sign in attempts log their DSI ID

### PR DESCRIPTION
**Pull request by tahb**
* This also corrects the fact that no logs using Rails.logger were going
to papertrail. This is because without being configured logs were going
to <environment>.log, which isn't sent to PT due to how logspout
collects and forwards logs on the hosted environments. This is desipite
the configuration we have for lograge which just concerns itself with
formatting and colouring.
* Group logging config together for clarity and remove prod config for log level which was being overwritten
screen shot 2018-10-01 at 17 43 30

### CODE REVIEW

File: app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb [hunk 1]
```diff
@@ -45,9 +47,4 @@ def identifier
   def selected_school_urn
     auth_hash.dig('extra', 'raw_info', 'organisation', 'urn')
   end
-
-  def log_debug_information
-    anonymised_email = identifier.gsub(/(.)./, '\1*')
```
Comment (tahb) [inline app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb]: Anonymising this email wouldn't have been useful in the cases we've had so far. We need to find the exact email address of a person looking to sign in to confirm they are the ones requesting support, favouring the DSI ID will keep it completely anonymous in our logs and the full ID value can only be used with access to DSI support console.

File: config/environments/production.rb [hunk 2]
```diff
@@ -50,13 +50,6 @@
   config.force_ssl = true
   config.ssl_options = { redirect: { exclude: ->(request) { request.path =~ /check/ } } }
 
-  # Use the lowest log level to ensure availability of diagnostic information
-  # when problems arise.
-  config.log_level = :debug
```
Comment (despo) [inline config/environments/production.rb]: 👍

File: config/environments/production.rb [hunk 3]
```diff
@@ -78,14 +71,13 @@
...
 
-  # Use default logging formatter so that PID and timestamp are not suppressed.
-  config.log_formatter = ::Logger::Formatter.new
-
-  # Use a different logger for distributed setups.
-  # require 'syslog/logger'
+  # Logging
+  config.log_level = :info
+  config.log_tags = [:request_id] # Prepend all log lines with the following tags.
+  config.logger = ActiveSupport::Logger.new(STDOUT)
```
Comment (despo) [inline config/environments/production.rb]: 👍

File: config/environments/staging.rb [hunk 4]
```diff
@@ -78,6 +71,11 @@
   # Send deprecation notices to registered listeners.
   config.active_support.deprecation = :notify
 
+  # Logging
+  config.log_level = :debug
```
Comment (despo) [inline config/environments/staging.rb]: Do we need `debug` on staging?

**Comment by tahb** [comment]
[Log the DSI ID for users who sign into our service](https://trello.com/c/umEkh2S6/434-log-the-dsi-id-for-users-who-sign-into-our-service)

**Pull request merged**